### PR TITLE
Adapt plugin

### PR DESCRIPTION
**Pull request by TimStewartJ**
Adds Delmar's new requested plugin (https://github.com/LibreTexts/metalc/issues/228).

Things that are still needed:

- ~~Needs proper input validation for the Adapt ID.~~
- ~~Probably needs a better icon since I just took a random picture from Google.~~ Now uses the adapt favicon.
- ~~Might want to also rename `plugindos.js`.~~ Both buttons are in `plugin.js` now.

### CODE REVIEW

File: src/scripts/plugindos.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const registerAdapt = () => {
+    CKEDITOR.plugins.add('libretextsAdapt', {
+      init(editor) {
+        // remove restrictions on what tags and css properties can be output
```
Comment (moorepants) [inline src/scripts/plugindos.js]: I was imagining not needing to to make a second plugin. Can't all the code inside init(editor){} just go in the original plugin.js file? i.e. there is a single plugin that registers two button.

Comment (moorepants) [reply src/scripts/plugindos.js]: Not that we should do this, but a single button could work too. There could be a radio button in the dialog to select "query" or "adapt".

Comment (TimStewartJ) [reply src/scripts/plugindos.js]: Ah I see what you mean, I'll see what I can do about consolidating the code a bit more.

Comment (TimStewartJ) [reply src/scripts/plugindos.js]: Alright, both are now in the same `registerPlugin` function in `plugin.js` for now. I'm not too sure how feasible or possible it is to have both be in `init(editor) {}`, but I'll try to look into reducing redundancy. Having a single button would definitely accomplish that, but I'm unsure of the implications of that from a UI/UX standpoint.

Comment (moorepants) [reply src/scripts/plugindos.js]: Keep it two buttons.

Comment (moorepants) [reply src/scripts/plugindos.js]: Don't fuss over the redundancy too much. Main thing is to get it working.

I was imagining that plugin != button, but maybe that's what plugin sorta means in ckeditor.

**Comment by sandertyu** [comment]
The plugin looks good to me. You can see it deployed [here](https://query.libretexts.org/Development/Query_Plugin_Demo). I'm not sure what input validation is needed, we are using the same method as done with the original query plugin. Of course, we'll still want Delmar to take a look.

Also, with the new plugin. I think we should rename this repository to something like "CKEditor-Additional-Plugins" maybe. I'm not sure if this would alter the behavior elsewhere.

**Comment by moorepants** [comment]
I've asked delmar for feedback.

I don't think we need to worry about renaming the plugin.

But please add information to the README explaining what this does, both query and adapt.

**Comment by TimStewartJ** [comment]
Alright, the input has now been updated as per Delmar's specifications (being a "(number-number) string"), and the validation works accordingly.

File: src/scripts/plugin.js [hunk 2]
```diff
@@ -51,18 +53,81 @@ const registerPlugin = () => {
...
+          contents: [
+            {
+              id: 'tab1',
+              label: 'First Tab',
+              title: 'First Tab Title',
+              accessKey: 'Q',
+              elements: [
+                {
+                  type: 'text',
+                  label: 'Please enter a valid Adapt ID',
```
Comment (moorepants) [inline src/scripts/plugin.js]:
```suggestion
                  label: 'Please enter a valid Adapt ID in the format 123-12345',
```

**Comment by sandertyu** [comment]
As far as I can tell, the only thing left now is to update the README then we are ready for release (merging to master from staging).

**Comment by sandertyu** [comment]
The README is mostly updated except I have not included a picture of the plugin in action. The current behavior seems very unenlightening:

adapt

If this is really all it is supposed to do (make the `Activity` appear hyperlinked?) then I can just add it in and we're done.

**Comment by moorepants** [comment]
Just leave the picture out, that's fine. I guess once we push to production we'll see what it does.

**Comment by sandertyu** [comment]
Gonna merge this to staging so it can be prepared for deployment

**Pull request merged**